### PR TITLE
Fix pkgmgr game install with RUN_IN_PLACE=0

### LINKED ISSUE #8083: pkgmgr install fails with RUN_IN_PLACE=0
Installing a game (eg: Mineclone2) results in the following message:

    Failed to install mineclone2 to /home/user/.minetest/games/mineclone2

### PR DESCRIPTION
**Pull request by pauloue**
Fixes #8083

### CODE REVIEW

File: src/script/lua_api/l_mainmenu.cpp [hunk 1]
```diff
@@ -844,22 +844,21 @@ bool ModApiMainMenu::isMinetestPath(std::string path)
 		return true;
 
 	/* games */
-	if (fs::PathStartsWith(path,fs::RemoveRelativePathComponents(porting::path_share + DIR_DELIM + "games")))
+	if (fs::PathStartsWith(path,fs::RemoveRelativePathComponents(porting::path_user + DIR_DELIM + "games")))
```
Comment (rubenwardy) [inline src/script/lua_api/l_mainmenu.cpp]: path_share is also a valid place for it to be, so I suggest duplicating this statement - one for share and one for user

Comment (pauloue) [reply src/script/lua_api/l_mainmenu.cpp]: I thought that method was to allow downloading to user directories only?

Comment (SmallJoker) [reply src/script/lua_api/l_mainmenu.cpp]: It is indeed only used to check whether the directory may be modified. However, I suggest renaming this function to `mayModifyPath` or similar.

File: src/script/lua_api/l_mainmenu.cpp [hunk 2]
```diff
@@ -838,28 +838,27 @@ int ModApiMainMenu::l_get_mainmenu_path(lua_State *L)
 }
 
 /******************************************************************************/
-bool ModApiMainMenu::isMinetestPath(std::string path)
+bool ModApiMainMenu::mayModifyPath(std::string path)
```
Comment (nerzhul) [inline src/script/lua_api/l_mainmenu.cpp]: can you fix the function signature which is wrong in the original code to `const std::string &path`

Thanks

Comment (pauloue) [reply src/script/lua_api/l_mainmenu.cpp]: done

File: src/script/lua_api/l_mainmenu.cpp (hunk 1 above)
Comment (ClobberXD) [reply src/script/lua_api/l_mainmenu.cpp]: It'd be nice if you could add a space after `path,`. Many such spaces are missing throughout the file, but at least the code you touched can be rid of code-style issues.

**Pull request merged**